integrations/slack/src/actions/queryAskAI.ts

```diff
@@ -149,59 +154,127 @@ async function getRelatedSources(params: {
+        threadId,
+        userId,
+        text,
+        messageType,
+        context,
+        authorization,
+        responseUrl,
+    } = params;
+    const { api } = context;
 
     const askText = `_Asking: ${stripMarkdown(text)}_`;
     logger.info(`${askText} (channelId: ${channelId}, teamId: ${teamId}, userId: ${userId})`);
 
-    const { client, installation } = await getInstallationApiClient(api, teamId);
+    const installation = await getIntegrationInstallationForTeam(context, teamId);
     if (!installation) {
         throw new Error('Installation not found');
     }
-    // Authenticate as the installation
+    // Use the slack access token
     const accessToken = (installation.configuration as SlackInstallationConfiguration)
         .oauth_credentials?.access_token;
 
     // strip a bot name if the user_id from the request is present in the query itself (specifically for a bot mention)
     // @ts-ignore
     const parsedQuery = stripMarkdown(stripBotName(text, authorization?.user_id));
 
-    // async acknowledge the request to the end user early
-    slackAPI(
-        context,
-        {
-            method: 'POST',
-            path: messageType === 'ephemeral' ? 'chat.postEphemeral' : 'chat.postMessage',
-            responseUrl,
-            payload: {
-                channel: channelId,
-                text: askText,
-                ...(userId ? { user: userId } : {}), // actually shouldn't be optional
-                ...(threadId ? { thread_ts: threadId } : {}),
+    await Promise.all([
+        // acknowledge the ask query back to the user
+        slackAPI(
+            context,
+            {
+                method: 'POST',
+                path: messageType === 'ephemeral' ? 'chat.postEphemeral' : 'chat.postMessage',
+                responseUrl,
+                payload: {
+                    channel: channelId,
+                    text: askText,
+                    ...(userId ? { user: userId } : {}), // actually shouldn't be optional
+                    ...(threadId ? { thread_ts: threadId } : {}),
+                },
+            },
+            {
+                accessToken,
             },
+        ),
+        // Queue a task to process the AskAI query asynchronously. Because workers have a 30s timeout on
+        // waitUntil which is not enough for scenarios where AskAI might take longer to respond.
+        queueQueryAskAI({
+            ...params,
+            accessToken,
+            installation,
+            query: parsedQuery,
+        }),
+    ]);
+}
+
+/**
+ * Queues an integration task to process the AskAI query asynchronously.
+ */
+async function queueQueryAskAI(
+    params: IQueryAskAI & {
+        query: string;
+        installation: IntegrationInstallation;
+        accessToken: string | undefined;
+    },
+) {
+    const { accessToken, installation, query, context, ...rest } = params;
+
+    // Authenticate as the integration
+    const gitbook = new GitBookAPI({
```

Comment:
Can't you get the client passed from the caller that could get it from the context?